Providers/Profile.swift

```diff
@@ -734,25 +737,17 @@ open class BrowserProfile: Profile {
             return
         }
         self.prefs.setTimestamp(now, forKey: PrefsKeys.PollCommandsTimestamp)
-        let accountManager = self.rustFxA.accountManager.peek()
-        accountManager?.deviceConstellation()?.pollForCommands { commands in
-            if let commands = try? commands.get() {
-                for command in commands {
-                    switch command {
-                    case .tabReceived(let sender, let tabData):
-                        // The tabData.entries is the tabs history
-                        // we only want the last item, which is the tab
-                        // to display
-                        let title = tabData.entries.last?.title ?? ""
-                        let url = tabData.entries.last?.url ?? ""
-                        if let json = try? accountManager?.gatherTelemetry() {
-                            let events = FxATelemetry.parseTelemetry(fromJSONString: json)
-                            events.forEach { $0.record(intoPrefs: self.prefs) }
-                        }
-                        if let url = URL(string: url) {
-                            self.syncDelegate?.displaySentTab(for: url, title: title, from: sender?.displayName)
+        self.rustFxA.accountManager.upon { accountManager in
+            accountManager.deviceConstellation()?.pollForCommands { commands in
+                if let commands = try? commands.get() {
+                    let urls = commands.compactMap { command in
+                        switch command {
+                        case .tabReceived(_, let tabData):
+                            let url = tabData.entries.last?.url ?? ""
+                            return URL(string: url)
                         }
                     }
+                    self.sendTabDelegate?.openSendTabs(for: urls)
```

Comment:
nit, easier to read with guard IMO
```suggestion
                guard let commands = try? commands.get() else { return }
                let urls = commands.compactMap { command in
                    switch command {
                    case .tabReceived(_, let tabData):
                        let url = tabData.entries.last?.url ?? ""
                        return URL(string: url)
                    }
                }
                self.sendTabDelegate?.openSendTabs(for: urls)
```